tests/pyfunc/test_pyfunc_schema_enforcement.py

```diff
@@ -1363,31 +1403,76 @@ def predict(self, context, model_input, params=None):
     ] == np.datetime64("2023-06-26 00:00:00")
 
 
-def test_enforce_schema_with_arrays_in_python_model_predict():
-    params = {
-        "int_array": np.array([np.int32(1), np.int32(2)]),
+def test_enforce_schema_in_python_model_serving(test_params_basic):
+    signature = infer_signature(["input1"], params=test_params_basic)
+    with mlflow.start_run():
+        model_info = mlflow.pyfunc.log_model(
+            python_model=PythonModelWithBasicParams(),
+            artifact_path="test_model",
+            signature=signature,
+        )
+
+    # params in payload should be json serializable
+    test_params = {
+        "str_param": "str_a",
+        "int_param": 1,
+        "bool_param": True,
+        "double_param": 1.0,
+        "float_param": 0.1,
+        "long_param": 100,
+        "datetime_param": datetime.datetime(2023, 6, 6, 0, 0, 0),
+        "str_list": ["a", "b", "c"],
+        "bool_list": [True, False],
         "double_array": np.array([1.0, 2.0]),
-        "float_array": np.array([np.float32(1.0), np.float32(2.0)]),
-        "long_array": np.array([1, 2]),
-        "datetime_array": np.array(
-            [np.datetime64("2023-06-26 00:00:00"), np.datetime64("2023-06-26 00:00:00")]
-        ),
     }
+    response = pyfunc_serve_and_score_model(
+        model_info.model_uri,
+        data=dump_input_data(["a", "b"], params=test_params),
+        content_type=pyfunc_scoring_server.CONTENT_TYPE_JSON,
+        extra_args=["--env-manager", "local"],
+    )
+    assert response.status_code == 200
+    prediction = json.loads(response.content.decode("utf-8"))["predictions"]
+    for param, value in test_params.items():
+        if param == "double_array":
+            assert (prediction[param] == value).all()
```

Comment:
Not a blocker, but when cheking a numpy array or pandas dataframe, we should use `np.testing.assert_array_equal` or `pandas.testting.assert_frame_equal`. `assert (prediction[param] == value).all()` tells us True or False, but `assert_xxx_yyy` functions gives more information which is helpful for debugging.